enumerables.rb

```diff
@@ -0,0 +1,332 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity, Lint/Void
+p '-----------|||||--------'
+p '----------{(o o)}--------'
+p '--------ooO-(_)-Ooo------'
+# Initial comment
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    array = to_a
+    size.times { |index| yield(array[index]) }
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    array = to_a
+    size.times { |index| yield(array[index], index) }
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = to_a
+    arraya = []
+    array.length.times do |a|
+      array[a]
+      arraya << array[a] if yield (array[a])
+    end
+    arraya
+  end
+
+  def my_all?(*arg)
```

Comment:
When no block or argument is given returns true when none of the collection members are false or nil.
![Screen Shot 2020-07-21 at 7 24 31 PM](https://user-images.githubusercontent.com/52430951/88120134-d9018380-cb87-11ea-93e3-ec2628b785ad.png)

Reply:
## specify pattern
Traceback (most recent call last):
enumerables.rb:364:in `<main>': undefined local variable or method `arr' for main:Object (NameError)
Did you mean?  Array